Mathlib/Data/Fin/Basic.lean

```diff
@@ -437,14 +437,16 @@ theorem last_pos' [NeZero n] : 0 < last n := NeZero.pos n
 
 theorem one_lt_last [NeZero n] : 1 < last (n + 1) := (lt_add_iff_pos_left 1).mpr (NeZero.pos n)
 
-theorem top_eq_last (n : ℕ) : ⊤ = Fin.last n :=
+theorem top_eq_last (n : ℕ) : ⊤ = last n :=
   rfl
 #align fin.top_eq_last Fin.top_eq_last
 
-theorem bot_eq_zero (n : ℕ) : ⊥ = (0 : Fin (n + 1)) :=
+theorem bot_eq_zero (n : ℕ) [NeZero n] : ⊥ = (0 : Fin n) :=
   rfl
 #align fin.bot_eq_zero Fin.bot_eq_zero
 
+theorem bot_eq_top_of_fin_one : (⊥ : Fin 1) = (⊤ : Fin 1) := rfl
```

Comment:
That's just `Subsingleton.elim`, right?

Reply:
It is, but it seemed useful to have the special case? Also I don't think this depends on the subsingleton result.

Reply:
I have removed this.